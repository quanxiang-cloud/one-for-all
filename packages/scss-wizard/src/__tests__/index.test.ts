jest.mock('../format');

import fs from 'fs';
import path from 'path';

import toSCSS from '../to-scss';
import toAST from '../to-ast';

test('convert_between_ast_and_scss', async () => {
  const scssStr = fs.readFileSync(path.join(__dirname, 'fixtures/index.scss'), { encoding: 'utf-8'});
  const ast = await toAST(scssStr);
  expect(ast.nodes).toMatchSnapshot();

  const scss = await toSCSS(ast);
  expect(scss).toMatchSnapshot();

  const _ast = await toAST(scss);
  expect(_ast.nodes).toEqual(ast.nodes);
});
