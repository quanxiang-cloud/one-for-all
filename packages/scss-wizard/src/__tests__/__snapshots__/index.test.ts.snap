// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert_between_ast_and_scss 1`] = `
Array [
  Object {
    "raws": Object {
      "before": "",
      "inline": true,
      "left": " ",
      "right": "",
      "text": "some unformatted scss",
    },
    "source": Object {
      "end": Object {
        "column": 24,
        "line": 1,
        "offset": 23,
      },
      "inputId": 0,
      "start": Object {
        "column": 1,
        "line": 1,
        "offset": 0,
      },
    },
    "text": "some unformatted scss",
    "type": "comment",
  },
  Object {
    "nodes": Array [
      Object {
        "prop": "color",
        "raws": Object {
          "before": "
  ",
          "between": ": ",
        },
        "source": Object {
          "end": Object {
            "column": 13,
            "line": 3,
            "offset": 46,
          },
          "inputId": 0,
          "start": Object {
            "column": 3,
            "line": 3,
            "offset": 36,
          },
        },
        "type": "decl",
        "value": "red",
      },
      Object {
        "name": "include",
        "params": "someMixin",
        "raws": Object {
          "afterName": " ",
          "before": "

  ",
          "between": "",
        },
        "source": Object {
          "end": Object {
            "column": 21,
            "line": 5,
            "offset": 69,
          },
          "inputId": 0,
          "start": Object {
            "column": 3,
            "line": 5,
            "offset": 51,
          },
        },
        "type": "atrule",
      },
      Object {
        "nodes": Array [
          Object {
            "prop": "font-size",
            "raws": Object {
              "before": " ",
              "between": ": ",
            },
            "source": Object {
              "end": Object {
                "column": 34,
                "line": 7,
                "offset": 105,
              },
              "inputId": 0,
              "start": Object {
                "column": 19,
                "line": 7,
                "offset": 90,
              },
            },
            "type": "decl",
            "value": "20px",
          },
        ],
        "raws": Object {
          "after": " ",
          "before": "

      ",
          "between": " ",
          "semicolon": true,
        },
        "selector": ".children",
        "source": Object {
          "end": Object {
            "column": 36,
            "line": 7,
            "offset": 107,
          },
          "inputId": 0,
          "start": Object {
            "column": 7,
            "line": 7,
            "offset": 78,
          },
        },
        "type": "rule",
      },
    ],
    "raws": Object {
      "after": "
",
      "before": "
",
      "between": "",
      "semicolon": false,
    },
    "selector": ".parent",
    "source": Object {
      "end": Object {
        "column": 1,
        "line": 8,
        "offset": 109,
      },
      "inputId": 0,
      "start": Object {
        "column": 1,
        "line": 2,
        "offset": 25,
      },
    },
    "type": "rule",
  },
]
`;

exports[`convert_between_ast_and_scss 2`] = `
"// some unformatted scss
.parent{
  color: red;

  @include someMixin;

      .children { font-size: 20px; }
}
"
`;
